app/controllers/application_controller.rb

```diff
@@ -1,8 +1,55 @@
 class ApplicationController < ActionController::API
   include JsonapiErrorsHandler
+  
+  class AuthorizationError < StandardError; end
+
+  rescue_from ::StandardError, with: lambda { |e| handle_error(e) }
+
+  rescue_from UserAuthenticatorService::AuthenticationError, with:
+    :authentication_error
+
+  rescue_from AuthorizationError, with:
+    :authorization_error
+
+  before_action :authorize!
 
   ErrorMapper.map_errors!(
       'ActiveRecord::RecordNotFound' => 'JsonapiErrorsHandler::Errors::NotFound'
   )
-  rescue_from ::StandardError, with: lambda { |e| handle_error(e) }
+
+  private
+
+  def authorize!
+    raise AuthorizationError unless current_user
+  end
+
+  def access_token
+    provided_token = request.authorization&.gsub("Bearer ", "")
+    @access_token = AccessToken.find_by(token: provided_token)
+  end
+
+  def current_user
+    @current_user = access_token&.user
```

Comment:
`&.` is a "safe navigation operator".  It allows the function to finish without error when concerning `nil`.  EX: callling `user` on `nil` results in error. Using the operator allows it to proceed without a `user`